vendor/github.com/moby/moby/pkg/ioutils/fswriters.go

```diff
@@ -11,7 +10,7 @@ import (
 // temporary file and closing it atomically changes the temporary file to
 // destination path. Writing and closing concurrently is not allowed.
 func NewAtomicFileWriter(filename string, perm os.FileMode) (io.WriteCloser, error) {
-	f, err := ioutil.TempFile(filepath.Dir(filename), ".tmp-"+filepath.Base(filename))
+	f, err := os.CreateTemp(filepath.Dir(filename), ".tmp-"+filepath.Base(filename))
```

Comment:
Can some explanation be provided for why this change is needed.

I see that the approach in the current version of moby is also to use `ioutil.Tempfile`, my question is, why does this need to change? Especially as a modification to a vendor package?

Reply:
Also, just looked at the original motivation (the CVE) and this does not appear to be a CVE about file management or temporary file issues, so this change seems unrelated?